raiden/transfer/channel.py

```diff
@@ -1868,6 +1912,54 @@ def handle_action_close(
     return TransitionResult(channel_state, events)
 
 
+def handle_action_coop_settle(
+    channel_state: NettingChannelState,
+    action_coop_settle: ActionChannelCoopSettle,
+    pseudo_random_generator: random.Random,
+    block_number: BlockNumber,
+) -> TransitionResult[NettingChannelState]:
+    events: List[Event] = list()
+
+    our_max_total_withdraw = get_max_withdraw_amount(
+        channel_state.our_state, channel_state.partner_state
+    )
+    partner_max_total_withdraw = get_max_withdraw_amount(
+        channel_state.partner_state, channel_state.our_state
+    )
```

Comment:
These only need to be computed if `valid_coop_settle` below is `True`.

Reply:
But to calculate `valid_coop_settle`we need `our_max_total_withdraw`. Your statement is still true about `partner_max_total_withdraw`, but this way both of the assignments are close together.

Reply:
Ah, right.